util/pkg/vfs/s3fs.go

```diff
@@ -447,7 +447,8 @@ func (p *S3Path) GetHTTPsUrl() (string, error) {
 		}
 		p.bucketDetails = bucketDetails
 	}
-	return fmt.Sprintf("https://%s.s3.%s.amazonaws.com/%s", p.bucketDetails.name, p.bucketDetails.region, p.Key()), nil
+	baseURL := fmt.Sprintf("https://%s.s3.%s.amazonaws.com", p.bucketDetails.name, p.bucketDetails.region)
+	return path.Join(baseURL, p.Key()), nil
```

Comment:
`fmt.Println(path.Join("https://example.com", "/path"))` returns `https:/example.com/path` which has too few `/` after the `https:`.

Reply:
Perhaps use a `URL` struct literal.

Reply:
Oh...sorry. I fixed it.
https://github.com/kubernetes/kops/pull/11649/commits/d52ec60c02473bfa886fbec36ecf60178d3fe3fc